cmake/linux/compilerconfig.cmake

```diff
@@ -61,7 +61,7 @@ if(CMAKE_VERSION VERSION_LESS 3.24.0)
     add_compile_options(-Werror)
   endif()
 else()
-  set(CMAKE_COLOR_DIAGNOSTICS ON)
+  set(CMAKE_COLOR_DIAGNOSTICS ON CACHE BOOL "Enable color diagnostics")
```

Comment:
As long as the build system itself does not exclude versions CMake 3.24 and older, then the ability to switch off color diagnostics needs to be added for that version as well.

Alternatively (and probably much easier) would be to remove that automatism entirely and have users set `CMAKE_COLOR_DIAGNOSTICS` by themselves if they need/want it (which means removing the entire conditional block).

Reply:
Ok, I will remove the color-diagnose settings entirely so that cmake will follow users' preference.